lib/claim/token-holder-revenue-fund-contracts-ensemble.js

```diff
@@ -0,0 +1,359 @@
+'use strict';
+
+const {utils: {bigNumberify}} = require('ethers');
+const TokenHolderRevenueFundContract = require('./token-holder-revenue-fund-contract');
+const NestedError = require('../nested-error');
+
+const _ensemble = new WeakMap();
+const _spannedByCurrency = new WeakMap();
+const _firstAccrualOffset = new WeakMap();
+
+const currencyToKey = (currency) => `${currency.ct.toString()}-${currency.id}`;
+
+const minBigNumber = function (n1, n2) {
+    return n1.lt(n2) ? n1 : n2;
+};
+
+/**
+ * @class TokenHolderRevenueFundContractsEnsemble
+ * An ensemble of instances of TokenHolderRevenueFundContract
+ * @alias module:nahmii-sdk
+ */
+class TokenHolderRevenueFundContractsEnsemble {
+    constructor(walletOrProvider, abstractionNames = ['TokenHolderRevenueFund'], firstAccrualOffset = 0) {
```

Comment:
Do we want to document the constructor?

Reply:
We don't have a consistent tradition for doing so. But surely it helps. I'll add a few lines of comment. 👍